### ISSUE TITLE
Quick install stalls at approving certificate on windows

### ISSUE THREAD

**Issue by nikovlyubomir** [ContainerSolutions/trow#80]
I am trying to install the trow, by first going to quick-install folder and then typing "./install.sh".
The installation seems to start fine, but at one moment i get stucked to something like this:

Approving certificate. This may take some time.
...................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................

I tried several times restarting the installation.

Thank you.

**Comment by amouat** [comment]
Which platform and version are you on e.g. GKE, minikube?

On some platforms certificate approval can take a while (low number of minutes), but judging by the number of dots, it's not happening for some reason. It may be a permissions thing.

**Comment by nikovlyubomir** [comment]
I am using minikube version: v1.9.2.

**Comment by amouat** [comment]
Yeah, don't wait more than 5 mins. I think I've seen this.

I *think* the problem is something like the first run got cancelled and the subsequent runs got confused about the old certificate. I'll try to debug it later. If you don't have anything valuable in the minikube vm, the easiest temporary workaround is to nuke your vm (minikube delete) and start from a fresh install. I ran it on minikube 1.9.2 yesterday and it works as long as you're not using the containerd driver (which you won't be as long as you didn't explicitly select it)

**Comment by amouat** [comment]
Just FYI, it took 1m40s for the approving certificate step on my last run with minikube. I'm not really sure why it takes so long, but we're waiting on k8s, not minikube.

**Comment by amouat** [comment]
I'm having trouble recreating this, could you provide me with the minikube start-up message e.g:

```
minikube start --vm-driver kvm2
😄  minikube v1.9.2 on Ubuntu 19.10
✨  Using the kvm2 driver based on existing profile
👍  Starting control plane node m01 in cluster minikube
🏃  Updating the running kvm2 "minikube" VM ...
🐳  Preparing Kubernetes v1.18.0 on Docker 19.03.8 ...
🌟  Enabling addons: default-storageclass, storage-provisioner
🏄  Done! kubectl is now configured to use "minikube"
```

And the output of `kubectl get csr` after running the install script.

**Comment by nikovlyubomir** [comment]
I just deleted minikube as you told me and started again. That's what I see now:

* minikube v1.9.2 on Microsoft Windows 10 Enterprise 10.0.18363 Build 18363
* Automatically selected the hyperv driver
* Starting control plane node m01 in cluster minikube
* Creating hyperv VM (CPUs=2, Memory=4000MB, Disk=20000MB) ...
* Preparing Kubernetes v1.18.0 on Docker 19.03.8 ...
! This VM is having trouble accessing https://k8s.gcr.io
* To pull new external images, you may need to configure a proxy: https://minikube.sigs.k8s.io/docs/reference/networking/proxy/
* Enabling addons: default-storageclass, storage-provisioner
* Done! kubectl is now configured to use "minikube"

I just realized this one "! This VM is having trouble accessing https://k8s.gcr.io", may it be the problem?

About kubectl get csr i do not find any resources ("No resources found.")

**Comment by amouat** [comment]
Oh, interesting. To be honest, I haven't tried on windows. 

It sounds like the actual problem is the `kubectl apply` line in the installer where it pulls the Trow images is failing. To fix this, you'll need to figure out why the VM doesn't have external connectivity (are you on a VPN?).

What I should do is check the status of the `kubectl apply` and throw an error if there's a problem.

Once you've fixed the connectivity issue, I think you may still hit a host routing issue. On Mac and Linux, this is solved by editing /etc/hosts, but I'm not sure if that will work on Windows. Are you using WSL?

**Comment by Cris01** [comment]
Hello, I have the same problem, it is eternally supplying the certificate, I am doing the installation in a cluster that I have in bare metal, in minikube it worked without problems.

I appreciate any tip that helps me solve the problem.

**Comment by amouat** [comment]
Thanks @Cris01. At the moment I'm really not sure you have the same issue though.

Could you open a new issue and include a few more details about your setup - at a minimum your version of Kubernetes and host operating system - and what commands you're running?

**Comment by amouat** [comment]
@nikovlyubomir how did you get on?

**Comment by Cris01** [comment]
Sorry for the lack of detail in my question, I already managed to solve the problem, the problem was that the pod could not write in the volume with user 33333, I changed the user id to a valid user id in the cluster and it has worked .

**Comment by Kribylet** [comment]
Ran into a hanging certificate approval on Ubuntu 18.04 using kubeadm. I ran kubectl delete according to the uninstall instructions. Afterwards, browsing the namespace for trow showed a remaining secret and configmap for trow. Deleting both and running the quick install approved instantly.

**Comment by amouat** [comment]
Closing in favour of #198.
